### PR TITLE
Update GitHub Actions for automating PR review

### PR DESCRIPTION
**Pull request by meteorcloudy**
The GitHub Action was implemented in https://github.com/bazelbuild/continuous-integration/pull/1887

This PR updated/created three workflows from the same GitHub Action, distinguished by `action-type`:
- `notify_maintainers`: Notify module maintainers whenever a PR is opened, updated or re-opened for the main branch with changes under "modules" directory.
- `dismiss_approvals`: Dismisses stale approvals whenever a PR is updated. We need this because the `Dismiss stale pull request approvals when new commits are pushed` branch protection from GitHub doesn't work for reviewers who is not a repo maintainer.
- `review_prs`: 
    -  Review all open PRs every 10 mins. 
    - Approve a PR if the latest change is approved by at least one module maintainer for each modified module. (Not yet) 
    - Merge the PR if presubmit passes. (I'm not turning on this for now in case there is still something we are missing, instead it will ping the bcr maintainers to take a final look.)

Fixes https://github.com/bazelbuild/bazel-central-registry/issues/130

### CODE REVIEW

**Comment by fmeum** [comment]
I like the new strategy, but I still see a potential issue: The PR approval will not be dismissed instantly since the action will take some time to run. By carefully timing the malicious push, it seems possible to get it in in time for the approval (and future merge) action but before the dismissal action has completed.

**Comment by meteorcloudy** [comment]
I'm experimenting with https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository. 

The idea is that we can wait until all workflow runs of `dismiss_approvals` to be completed before reviewing the PRs, does that should good to you? 

> I think we should not enable automatic merge right now to prevent the toctou scenario mentioned.

Yes, I'm not enabling auto-merge anyway. Just pinging the BCR maintainers to take a final look after approving. https://github.com/bazelbuild/continuous-integration/blob/a4d362a00e617f2f99250cf03f5476c52b85d65f/actions/bcr-pr-reviewer/index.js#L271-L282

**Comment by mihaimaruseac** [comment]
Yeah, this seems good. Thank you!

**Review by fmeum — APPROVED** [review]

**Pull request merged**
